packages/basebridge/src/BaseBridge.ts

```diff
@@ -0,0 +1,207 @@
+import {
+  BASE_ERC20_ABI,
+  BASE_ERC20_BRIDGE_ADDRESS,
+  BASE_ETH_ABI,
+  BASE_ETH_BRIDGE_ADDRESS,
+  ETH,
+  ETHEREUM_ERC20_ABI,
+  ETHEREUM_ERC20_BRIDGE_ADDRESS,
+  ETHEREUM_ETH_ABI,
+  ETHEREUM_ETH_BRIDGE_ADDRESS,
+  baseContracts,
+  ethereumContracts,
+} from './constants'
+import {
+  type BridgeActionParams,
+  type TransactionFilter,
+  compressJson,
+} from '@rabbitholegg/questdk'
+import { CHAIN_TO_TOKENS, Chains } from '@rabbitholegg/questdk-plugin-utils'
+import { type Address } from 'viem'
+
+export const bridge = async (
+  params: BridgeActionParams,
+): Promise<TransactionFilter> => {
+  const { sourceChainId } = params
+
+  return sourceChainId === Chains.ETHEREUM
+    ? bridgeFromEthereum(params)
+    : bridgeFromBase(params)
+}
+
+const bridgeFromEthereum = async (
+  params: BridgeActionParams,
+): Promise<TransactionFilter> => {
+  const {
+    amount,
+    recipient,
+    destinationChainId,
+    sourceChainId,
+    contractAddress,
+    tokenAddress,
+  } = params
+
+  if (destinationChainId !== Chains.BASE || sourceChainId !== Chains.ETHEREUM) {
+    return failingFilter()
+  }
+
+  if (
+    contractAddress !== undefined &&
+    !ethereumContracts.includes(contractAddress)
+  ) {
+    return failingFilter()
+  }
+
+  if (tokenAddress === ETH) {
```

Comment:
This will be an issue as most of the time tokenAddress will be undefined (when any token is valid)

Reply:
```suggestion
const bridgeFromEthereum = async (
  params: BridgeActionParams,
): Promise<TransactionFilter> => {
  const {
    amount,
    recipient,
    destinationChainId,
    sourceChainId,
    contractAddress,
    tokenAddress,
  } = params

  if (destinationChainId !== Chains.BASE || sourceChainId !== Chains.ETHEREUM) {
    return failingFilter()
  }

  if (
    contractAddress !== undefined &&
    !ethereumContracts.includes(contractAddress)
  ) {
    return failingFilter()
  }

  return compressJson({
    chainId: Chains.ETHEREUM,
    to: { $or: [ETHEREUM_ETH_BRIDGE_ADDRESS, ETHEREUM_ERC20_BRIDGE_ADDRESS] },
    input: {
      $or: [
        { $abi: ETHEREUM_ETH_ABI, _to: recipient, _value: amount },
        {
          $abi: ETHEREUM_ERC20_ABI,
          mintRecipient: recipient,
          amount,
          burnToken: tokenAddress,
        },
      ],
    },
  })
}
```

Reply:
On further inspection it seems to work fine 👍